### PR TITLE
Deprecate tskit module attributes

### LINKED ISSUE #991: Remove tskit compatability layer?
To avoid breaking people's code while we split msprime from tskit, we have a compatibility shim in ``__init__.py`` which imports tskit classes into the msprime namespace. So, for people who used things like ``msprime.TableCollection`` their code would still work.

For 1.0, we need to decide whether we keep supporting this or not. The pros for removing are, we'll need to do it at *some* point, and if we don't do it at 1.0, then when can we do it? The break from tskit was at version 0.7.

The cons are that we will break some code that's out there. I don't think there's a lot of code out there that uses ``msprime.TableCollection`` etc though, so I'd be surprised if it broke much.

#990 removed all dependence on old classes like msprime.TableCollection, etc from the test suite.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #991

### CODE REVIEW

File: tests/test_deprecation.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            "pack_strings",
+            "pack_bytes",
+            "unpack_bytes",
+            "unpack_strings",
+            "validate_provenance",
+            "NODE_IS_SAMPLE",
+            "FORWARD",
+            "REVERSE",
+        ]:
+            print(name)
```
Comment (jeromekelleher) [inline tests/test_deprecation.py:70]: print

**Review by jeromekelleher — APPROVED** [review]
Yep, LGTM. I guess it's *slightly* more general because it allows people to import new stuff from tskit as well, but since it's throwing a warning that seems harmless.

**Review by petrelharp — COMMENTED** [review]
Looks good!

File: tests/test_deprecation.py (hunk 1 above)
Comment (benjeffery) [reply tests/test_deprecation.py]: Dang it. Fixed. If anyone knows how to set a breakpoint that only fires on failing tests that would be helpful!

**Comment by benjeffery** [comment]
Fixed up and PR number added to changelog.

**Pull request merged**